lib/grpc/client/interceptors/logger.ex

```diff
@@ -18,40 +17,49 @@ defmodule GRPC.Client.Interceptors.Logger do
       {:ok, channel} = GRPC.Stub.connect("localhost:50051", interceptors: [{GRPC.Client.Interceptors.Logger, level: :warning}])
   """
 
-  require Logger
-
   @behaviour GRPC.Client.Interceptor
 
-  @impl true
+  require Logger
+
+  @impl GRPC.Client.Interceptor
   def init(opts) do
     level = Keyword.get(opts, :level) || :info
     [level: level]
   end
 
-  @impl true
+  @impl GRPC.Client.Interceptor
   def call(%{grpc_type: grpc_type} = stream, req, next, opts) do
+    {time, result} = :timer.tc(next, [stream, req])
```

Comment:
Why do you think a try is necessary here? And what about using telemetry instead?

Reply:
Because otherwise you won't get any logs for the call anymore. In the previous version you would get the starting but not the terminal log.

I think telemetry would be an addition, but doesn't substitute the logs

Reply:
Ah ok, I get you. I'll improve it.

Related to telemetry what I mean is decoupling the call from the interceptor. In the same way phoenix instruments traces (see opentelemetry_phoenix). It will allow clients to make structured logging as well. Basically you emit an event instead of logging and we subscribe in the interceptor to the event, then log in the handle function.

Reply:
I think having the log message coupled here is the more natural way, although a telemetry event being published alongside it wouldn't hurt.

Reply:
I've applied your suggestions regarding safer logs with try/rescue keeping the same way we were measuring time, without timex.